test/e2e/migrationhierarchy_test.go

```diff
@@ -0,0 +1,341 @@
+package e2e
+
+import (
+	danav1 "github.com/dana-team/hns/api/v1"
+	. "github.com/dana-team/hns/pkg/testutils"
+	. "github.com/onsi/ginkgo/v2"
+)
+
+var _ = Describe("MigrationHierarchy", func() {
+	nsRoot := GenerateE2EName("root")
+
+	BeforeEach(func() {
+		CleanupTestNamespaces()
+		CleanupTestMigrationHierarchies()
+
+		// set up root namespace
+		CreateRootNS(nsRoot, rqDepth)
+		CreateResourceQuota(nsRoot, nsRoot, storage, "100Gi", cpu, "100", memory, "100Gi", pods, "100", gpu, "100")
+	})
+
+	AfterEach(func() {
+		CleanupTestNamespaces()
+		CleanupTestMigrationHierarchies()
+	})
+
+	It("should migrate subnamespace that have a CRQ or its direct parent have a CRQ,", func() {
+		nsA := GenerateE2EName("a")
+		nsB := GenerateE2EName("b")
+		nsC := GenerateE2EName("c")
+		nsD := GenerateE2EName("d")
+		nsE := GenerateE2EName("e")
+		nsF := GenerateE2EName("f")
+
+		// create hierarchy
+		CreateSubnamespace(nsA, nsRoot, false, storage, "50Gi", cpu, "50", memory, "50Gi", pods, "50", gpu, "50")
+		CreateSubnamespace(nsB, nsA, false, storage, "25Gi", cpu, "25", memory, "25Gi", pods, "25", gpu, "25")
+		CreateSubnamespace(nsC, nsB, false, storage, "10Gi", cpu, "10", memory, "10Gi", pods, "10", gpu, "10")
+		CreateSubnamespace(nsD, nsC, false, storage, "5Gi", cpu, "5", memory, "5Gi", pods, "5", gpu, "5")
+		CreateSubnamespace(nsE, nsC, false, storage, "5Gi", cpu, "5", memory, "5Gi", pods, "5", gpu, "5")
+		CreateSubnamespace(nsF, nsD, false, storage, "1Gi", cpu, "1", memory, "1Gi", pods, "1", gpu, "1")
+
+		CreateMigrationHierarchy(nsF, nsE)
+
+		// make sure the subnamespace was migrated and the parent was updated
+		FieldShouldContain("subnamespace", nsE, nsF, ".metadata.namespace", nsE)
+		FieldShouldContain("namespace", "", nsF, ".metadata.labels", danav1.Parent+":"+nsE)
+
+		// add test label to the migrated ns
+		LabelTestingNs(nsF)
```

Comment:
Open an issue about changing the behaviour of the migration labels setting so that we append labels instead of copying new ones; the current behavior can be harmful if there's a label on the namespace and it's then migrated - it would just be deleted

Reply:
opened #62